Assets/Scripts/Battlefield/CreatureScripts/UniqueCreature.cs

```diff
@@ -70,28 +76,72 @@ public void hightlight()
             outline.enabled = true;
         }
 
-        private void Update()
+        private void LateUpdate()
         {
-            if (highlightColor == 3 && Time.time > b)
+            if (start)
             {
-                //Glow(2);
+                currentTile = gridHolder.tiles[Mathf.RoundToInt(brain.startCoordinates.x), Mathf.RoundToInt(brain.startCoordinates.y)];
+                Move(currentTile);
+                start = false;
             }
+            
+        }
 
+        private void Update()
+        {
             if (!brain.isMyTurn && Time.time > b)
             {
                 outline.enabled = false;
             }
 
+            if (path != null)
+            {
+                MoveAlongPath(); 
+            } else
+            {
+                lr.enabled = false;
+            }
+
+
+
             animator.SetFloat("Speed", (agent.velocity.magnitude / 3.5f));
         }
 
+        private void MoveAlongPath()
+        {
+
+            Move(path[tileOnPath]);
+            onMoveTile = onTile(.1f);
+            if (path != null && onMoveTile && tileOnPath > 0)
+            {
+                tileOnPath--;
+            }
+            else if (path != null && onMoveTile && tileOnPath == 0)
+            {
+                path = null;
+            }
+        }
```

Comment:
Movement should not be determined by a the character but instead an abstraction like MovementSystem which gets the agent, and startPoint and destination. This here is writing the movement system into a unit itself which is just way too coupled and makes the class huge.